scripts/hab_conf_transfer_to_Q.Rmd

```diff
@@ -0,0 +1,129 @@
+-  Import the `habitat_confirmations.xls`
+
+```{r def-var, echo = TRUE}
+
+dir_project <- 'sern_skeena_2023'
+file_in <- 'scripts/hab_conf_transfer_to_Q.Rmd'
+# This assumes we use the same directory for the output file as the input file
+file_out <- 'hab_conf_transfer_to_Q.html'
+
+## Import the raw form_fiss_2023.gpkg
+form_fiss_site_raw <- fpr::fpr_sp_gpkg_backup(
+  path_gpkg = paste0("~/Projects/gis/", dir_project, '/data_field/2023/form_fiss_site_2023.gpkg'),
+  update_utm = TRUE,
+  update_site_id = FALSE,
+  write_back_to_path = FALSE,
+  write_to_csv = FALSE,
+  write_to_rdata = FALSE,
+  return_object = TRUE)
+
+
+## Import the habitat_confirmations.xlsm
+habitat_confirmations <- fpr::fpr_import_hab_con(path = '~/Projects/repo/fish_passage_skeena_2023_reporting/data/habitat_confirmations.xls', row_empty_remove = T)
+```
+
+## Extract temp, cond, and turb data
+
+We need to extract temp, cond, turb data from `step_2_fish_coll_data` sheet in the `habitat_confirmations.xls` file. 
+
+As per [PR feedback](https://github.com/NewGraphEnvironment/fish_passage_skeena_2023_reporting/pull/68#discussion_r1583971038), I used `all_of()` instead of `c()` to select columns, but turns out you still need to use `c()` within `all_of()` to select columns if the column names are not stored in variables or in a vector, so I made a vector of column names. Not sure if this is worth it because we only use this vector once, but may be nice to reproducibility in the future.
+
+```{r extract_temp_cond_turb, echo=TRUE, include = TRUE, eval = FALSE}
+vars <- c('local_name', 'temperature_c', 'conductivity_m_s_cm', 'turbidity')
+
+fish_coll_data <- habitat_confirmations %>%
+  purrr::pluck("step_2_fish_coll_data") %>%
+  dplyr::select(dplyr::all_of(vars))
+```
+
+
+## Join extracted data to form_fiss_site_2023
+
+Add temp, cond, and turb data from `step_2_fish_coll_data` to `form_fiss_site_2023`.
+
+Here we use `all_of(c())` to deselect the target columns. We can't use vars because is includes the `local_name` which we don't want to remove. We could have made another vector of column names, but since its only used once I didn't think it was necessary, but please let me know if I should. `all_of(c())` will still throw errors if a column is missing.
+
+As explained here https://github.com/NewGraphEnvironment/fish_passage_skeena_2023_reporting/issues/64, the three downstream sites at Thompson Creek were sampled twice (two haul passes) and each haul pass has there own temp, cond, and turb data. Therefore in `form_fiss_prep1` there are 3 extra observations (68) compared to `form_fiss_site_raw` (65), because `123377_ds_ef1`, `123377_ds_ef2`, `123377_ds_ef3` are duplicated to represent each haul pass. There is no haul pass column in `form_fiss_site_2023` so when doing QA read the comments to identify which fiss site corresponds to each haul pass (we should think about adding a haul pass column to `form_fiss_site_2023` in the future).
```

Comment:
`form_fiss_site_2023` may not be the place for electrofishing specifications.  More likely it will be `form_fish_sample` as discussed a little bit here https://github.com/NewGraphEnvironment/dff-2022/issues/12

Reply:
okay yes form_fish_sample sounds much more appropriate